core/controllers/suggestion.py

```diff
@@ -342,13 +350,49 @@ def get(self, target_type, suggestion_type):
             target_type: str. The type of the suggestion target.
             suggestion_type: str. The type of the suggestion.
         """
+        topic_name = self.request.get('topic_name', None)
+
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
+
+        opportunity_summary_exp_ids = []
+
+        if topic_name is not None:
+            topic = topic_fetchers.get_topic_by_name(topic_name)
+            if topic is not None:
+                exploration_opportunity_summaries = (
+                    opportunity_services.
+                    get_exploration_opportunity_summaries_by_topic_id(
+                    topic.id))
+
+                opportunity_summary_exp_ids = [
+                    opportunity.id for opportunity
+                    in exploration_opportunity_summaries]
+
         suggestions = suggestion_services.get_reviewable_suggestions(
-            self.user_id, suggestion_type)
+            self.user_id, suggestion_type, opportunity_summary_exp_ids)
         self._render_suggestions(target_type, suggestions)
 
 
+class AllTopicNamesHandler(base.BaseHandler):
```

Comment:
Why do we need this? It seems like it was copied from `contributor_dashboard.py`.

Reply:
Yep, I added this function as I was doing some testing. I have removed it now. 

Just for context, actually, this was supposed to be a draft PR (& therefore not ready for review) which is why I committed this function too. But someone modified the commit message at the top and the "Please do not review" text got overwritten. But thanks for noticing this! :)